api/src/mutations/mint.rs

```diff
@@ -132,6 +123,100 @@ impl Mutation {
         };
 
         let collection_mint_model = collection_mint_active_model.insert(conn).await?;
+
+        let mut collection_am = collections::ActiveModel::from(collection.clone());
+        collection_am.total_mints = Set(edition);
+        collection_am.update(&tx).await?;
+
+        // inserts a mint histories record in the database
+        let mint_history_am = mint_histories::ActiveModel {
+            mint_id: Set(collection_mint_model.id),
+            wallet: Set(input.recipient.clone()),
+            collection_id: Set(collection.id),
+            tx_signature: Set(None),
+            status: Set(CreationStatus::Pending),
+            created_at: Set(Utc::now().into()),
+            ..Default::default()
+        };
+
+        mint_history_am.insert(&tx).await?;
+
+        if collection.blockchain == BlockchainEnum::Solana {
+            let collection_metadata_json =
+                metadata_jsons::Entity::find_by_id(collection_mint_model.id)
+                    .one(conn)
+                    .await?
+                    .ok_or(Error::new("metadata json not found"))?;
+
+            let creators = mint_creators::Entity::find()
+                .filter(mint_creators::Column::CollectionMintId.eq(collection_mint_model.id))
+                .all(conn)
+                .await?;
+
+            let files = metadata_json_files::Entity::find()
+                .filter(metadata_json_files::Column::MetadataJsonId.eq(collection_metadata_json.id))
+                .all(conn)
+                .await?;
+            let attributes = metadata_json_attributes::Entity::find()
+                .filter(
+                    metadata_json_attributes::Column::MetadataJsonId
+                        .eq(collection_metadata_json.id),
+                )
+                .all(conn)
+                .await?;
+
+            let mut metadata_json_am = metadata_jsons::ActiveModel::from(collection_metadata_json);
+
+            metadata_json_am.id = Set(collection_mint_model.id);
+
+            let collection_mint_metadata_json = metadata_json_am.insert(&tx).await?;
+
+            // TODO: Look into bug regarding creators not being saved to the database when minting editions
```

Comment:
@imabdulbasit in refactoring this I saw no reference to saving creators for the collection mint when doing mint edition. Is this a bug we need to backfill?

Reply:
https://github.com/holaplex/hub-nfts/issues/241